go/vt/topo/keyspace.go

```diff
@@ -191,9 +200,55 @@ func (ts *Server) FindAllShardsInKeyspace(ctx context.Context, keyspace string,
 		opt.Concurrency = 1
 	}
 
+	// First try to get all shards using List if we can.
+	shardsPath := path.Join(KeyspacesPath, keyspace, ShardsPath)
+	listResults, err := ts.globalCell.List(ctx, shardsPath)
+	if err == nil { // We have everything we need to build the result
+		result := make(map[string]*ShardInfo, len(listResults))
+		for _, entry := range listResults {
+			// The shard key looks like this: /vitess/global/keyspaces/commerce/shards/-80/Shard
+			shardKey := string(entry.Key)
+			shardName := path.Base(path.Dir(shardKey)) // The base part of the dir is "-80"
+			// Validate the extracted shard name.
```

Comment:
nit picking here too, it might read better if we extract the content of this entire if block. it seems like we only need to send in `listResults` as an argument and return `results, err`, something like:
```go
if err == nil {
	return handleResults(listResults)
}
```

Reply:
Not sure it's more readable, but: https://github.com/vitessio/vitess/pull/15047/commits/3fb3e0421146a117fc2979b148265ed365adf9df